File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    while True:
        min_s = heapq.heappop(scoville)
        if min_s >= K:
            return answer 
        
        if not scoville:
            if min_s <K:
                return -1 
            break 
        min_s2 = heapq.heappop(scoville)
        heapq.heappush(scoville,min_s+min_s2*2)
        answer +=1
    return -1

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> queue = new PriorityQueue<>();
        
        for (int scov : scoville){
            queue.add(scov);
        }
                
        while (true){
            if (queue.size()==1){
                if (queue.peek() < K){
                    return -1;
                }
                break;
            }
                        
            Integer first = queue.poll();
            if (first >= K){
                return answer;
            }
            
            Integer second = queue.poll();
            
            queue.add(first + second*2);
            answer +=1 ;
            
        }
        
        return answer;
    }
}